editor/editor_properties_array_dict.cpp

```diff
@@ -321,100 +357,56 @@ void EditorPropertyArray::update_property() {
 			paginator = memnew(EditorPaginator);
 			paginator->connect("page_changed", callable_mp(this, &EditorPropertyArray::_page_changed));
 			vbox->add_child(paginator);
-		} else {
-			// Bye bye children of the box.
-			for (int i = property_vbox->get_child_count() - 1; i >= 0; i--) {
-				Node *child = property_vbox->get_child(i);
-				if (child == reorder_selected_element_hbox) {
-					continue; // Don't remove the property that the user is moving.
-				}
+		}
 
-				child->queue_free(); // Button still needed after pressed is called.
-				property_vbox->remove_child(child);
-			}
+		// Ensure there is the right number of slots.
+		int slot_number = MIN(size - offset, page_length);
+		while (slots.size() < slot_number) {
+			create_new_property_slot();
 		}
+		for (Slot slot : slots.slice(slot_number)) {
+			slot.hbox->queue_free();
+		}
```

Comment:
Alternatively you could do
```suggestion
		while (property_vbox->get_child_count() > slot_number) {
			memfree(slot.hbox);
		}
```
Though not sure which one is better.
You could at least replace `queue_free()` with `memfree()`; it's more common in the engine code.

Reply:
I'll replace the `queue_free` but I'm not sure how your loop is working as it does not seem to be changing the value of slot therefore freeing the same instance.